presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrder.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import com.facebook.presto.common.type.Type;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ZOrder
+{
+    private ZOrder()
+    {
+    }
+
+    public static int highestSetBit(int value)
+    {
+        int count = 0;
+        while (value != 0) {
+            value >>= 1;
+            count++;
+        }
+        return count;
+    }
+
+    public static char[] encodeIntegers(List<Integer> input)
+    {
+        // Find highest set bit within input integers
+        int bitCounter = 0;
+        for (int idx = 0; idx < input.size(); idx++) {
+            bitCounter |= input.get(idx);
+        }
+        // Position of highest set bit is maxBitLength
+        int maxBitLength = highestSetBit(bitCounter);
+        // Pad smaller numbers with 0s to the same length as maxBitLength
+        char[] address = new char[maxBitLength * input.size()];
```

Comment:
a Char in Java is 2 bytes. here I guess you need  (maxBitLength / 8 + 1) * input.size(), and the data type should be byte[].

Reply:
My current implementation stores only one bit in each char / byte. This definitely wastes some space, but makes indexing easier. I've converted the char arrays to byte arrays to save some space, and I'll work on squeezing 8 bits into each byte.